tests/chart_tests/test_astronomer_registry.py

```diff
@@ -8,20 +8,49 @@
     "kube_version",
     supported_k8s_versions,
 )
-def test_astronomer_registry_statefulset(kube_version):
-    """Test that helm renders a good statefulset template for astronomer
-    registry."""
-    docs = render_chart(
-        kube_version=kube_version,
-        show_only=["charts/astronomer/templates/registry/registry-statefulset.yaml"],
-    )
+class TestRegistryStatefulset:
+    def test_astronomer_registry_statefulset(self, kube_version):
+        """Test that helm renders a good statefulset template for astronomer
+        registry."""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=[
+                "charts/astronomer/templates/registry/registry-statefulset.yaml"
+            ],
+        )
 
-    assert len(docs) == 1
-    doc = docs[0]
-    assert doc["kind"] == "StatefulSet"
-    assert doc["apiVersion"] == "apps/v1"
-    assert doc["metadata"]["name"] == "release-name-registry"
-    assert any(
-        "quay.io/astronomer/ap-registry:" in item
-        for item in jmespath.search("spec.template.spec.containers[*].image", doc)
-    )
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "StatefulSet"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "release-name-registry"
+        assert any(
+            "quay.io/astronomer/ap-registry:" in item
+            for item in jmespath.search("spec.template.spec.containers[*].image", doc)
+        )
+
+    def test_astronomer_registry_statefulset_with_customenv(self, kube_version):
+        """Test that helm renders statefulset template for astronomer
+        registry with custom env values."""
+        docs = render_chart(
+            kube_version=kube_version,
+            values={
+                "astronomer": {
+                    "registry": {
+                        "extraEnv": [{"name": "REGISTRY_ENV_STORAGE", "value": "test"}]
+                    }
+                }
+            },
+            show_only=[
+                "charts/astronomer/templates/registry/registry-statefulset.yaml"
+            ],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "StatefulSet"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "release-name-registry"
+        assert {"name": "REGISTRY_ENV_STORAGE", "value": "test"} in doc["spec"][
+            "template"
+        ]["spec"]["containers"][0]["env"]
```

Comment:
I think something like this is better. I like this pattern we've used elsewhere where we define the object, then include it in the values, and then check for it. This makes it easy to change the object, and ensures it's the same input and output. Also using an env var name that is obviously a test env var will make it easy to spot if we need to debug it.

```suggestion
            extra_env = {"name": "TEST_ENV_VAR_876", "value": "test"}
            values={
                "astronomer": {
                    "registry": {
                        "extraEnv": [extra_env]
                    }
                }
            },
            show_only=[
                "charts/astronomer/templates/registry/registry-statefulset.yaml"
            ],
        )

        assert len(docs) == 1
        doc = docs[0]
        assert doc["kind"] == "StatefulSet"
        assert doc["apiVersion"] == "apps/v1"
        assert doc["metadata"]["name"] == "release-name-registry"
        assert extra_env in doc["spec"]["template"]["spec"]["containers"][0]["env"]
```